File: app/src/main/java/com/dionpapas/drink_more_water/database/WaterEntryDAO.java
package com.dionpapas.drink_more_water.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface WaterEntryDAO {
    @Query("SELECT * FROM DailyWaterEntries ORDER BY updated_at")
    LiveData<List<WaterEntry>> getAllWaterEntries();

    @Insert
    void insertWaterEntry(WaterEntry waterEntry);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateWaterEntry(WaterEntry waterEntry);

    @Delete
    void deleteWaterEntry(WaterEntry waterEntry);
}

File: app/src/main/java/com/dionpapas/drink_more_water/database/WaterEntryDAO.kt
package com.dionpapas.drink_more_water.database;

import androidx.lifecycle.LiveData;
import androidx.room.*
import java.util.List

@Dao
interface WaterEntryDAO {

    @Query("SELECT * FROM DailyWaterEntries ORDER BY updated_at")
    fun getAllWaterEntries(): LiveData<List<WaterEntry>>

    @Insert
    fun insertWaterEntry(waterEntry:WaterEntry)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updateWaterEntry(waterEntry:WaterEntry)

    @Delete
    fun deleteWaterEntry(waterEntry: WaterEntry)

    @Query("SELECT id,counter,updated_at FROM DailyWaterEntries ORDER BY updated_at")
    fun getTodaysValue(): LiveData<WaterEntry>

}